core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/FcJsonRpc.java

```diff
@@ -102,6 +97,91 @@ public boolean filecoinWalletHas(@JsonRpcParam("address") final String address)
     return fcSigners.availableIdentifiers().contains(address);
   }
 
+  @JsonRpcMethod("Filecoin.WalletSignMessage")
+  public FilecoinSignedMessage filecoinSignMessage(
+      @JsonRpcParam("identifier") final String identifier,
+      @JsonRpcParam("message") final FilecoinMessage message) {
+
+    final Bytes cborEncodedBytes = cborEncode(message);
+    final Bytes messageHash = Blake2b.sum256(cborEncodedBytes);
+    final Bytes encodedHashAndCode = encodeHashWithCode(messageHash);
+    final Bytes fcCid = createFilecoinCid(encodedHashAndCode);
+
+    final FilecoinSignature signature =
+        filecoinWalletSign(identifier, Bytes.wrap(fcCid).toBase64String());
+
+    return new FilecoinSignedMessage(message, signature);
+  }
+
+  private Bytes cborEncode(final FilecoinMessage message) {
```

Comment:
It might be a good idea to refactor these conversion methods out to FilecoinCborUtil or similar ... which may allow it to be independently unit tested.

Reply:
agreed - but unit testing is LARGELY pointless, as it needs to align with what Lotus does, which I don't FEEL is easy to document.